### PR TITLE
WASM support

### PR DESCRIPTION
**Pull request by DevSrSouza**
This PR adds the support for Kotlin WASM and update the Compose to `1.6.0-rc02`.

Testing WASM support:
`./gradlew :samples:multiplatform:wasmJsBrowserDevelopmentRun`

### CODE REVIEW

**Comment by likeich** [comment]
Does this need any testing? I would love to help in any way to get this implemented sooner.

**Comment by DevSrSouza** [comment]
Everything is working pretty well so far, State restauration, etc. Just need to fix the build and migrate the Kotlin DSL to the new structure.

Feel free to checkout to the branch and test by publishing locally and using in your project!

**Comment by joreilly** [comment]
You might have seen it already but just fyi about 1.8.0-RC2 of Kotlinx Coroutines....I _think_ it should mean you won't have to include atomicfu as dependency (https://github.com/Kotlin/kotlinx.coroutines/releases/tag/1.8.0-RC2)

**Comment by DevSrSouza** [comment]
Thanks @joreilly , I will update the Coroutines here and see if is not required anymore. 
But the plugin maybe should be required for be more performant, their is a [branch of Compose Multiplatform Core](https://github.com/JetBrains/compose-multiplatform-core/commit/2ac17ff1bdd0f20d479cac1be8b58dea4b1ce219) applying AtomicFu Gradle Plugin. I'm following the same approach to avoid boxing types when is possible.

**Comment by luludevmuniz** [comment]
Hey there, I would like to test it but I don't know how to publish it locally.
Is there a guide for it, or any plans to release it as a alpha?

File: samples/multiplatform/src/wasmJsMain/resources/index.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Compose App</title>
+    <script type="application/javascript" src="skiko.js"></script>
+    <script type="application/javascript" src="composeApp.js"></script>
```
Comment (programadorthi) [inline samples/multiplatform/src/wasmJsMain/resources/index.html:7]: Can we put these scripts before `</body>` tag? Loading scripts inside `head` tag is not recommended.

**Comment by programadorthi** [comment]
@luludevmuniz try `./gradlew -Pversion=1.0.0 publishToMavenLocal`

**Comment by joreilly** [comment]
When is it hoped to merge this and include in a release?

**Comment by Syer10** [comment]
Probably after Compose 1.6 is released

**Pull request merged**